### DISCUSSION TITLE
configuration des vlan depuis les interfaces

### DISCUSSION THREAD

**Discussion by csmith35-ctd** [netbox-community/netbox#18086] [Help Wanted!] [answered]
Hi, I'd like to generate the vlan list from the device interfaces.
I can easily retrieve the vlans with these commands

**Template**
```jinja
########################
{%- for interface in device.interfaces.all() -%}
  {%- if interface.enabled == true and interface.untagged_vlan.vid != “None” and interface.type != “virtual” %}
  Untagged VLANs: {{ interface.untagged_vlan.vid }}
  {% for vlan in interface.tagged_vlans.all() -%}
  {% if vlan.id != “” -%}
  Tagged VLANs: {{ vlan.vid }}
  {% endif -%}
  {%- endfor -%}
{%- endif -%}
{%- endfor %}
########################
``` 
**Rendered Config**
```
########################
  Untagged VLANs: 30
  Untagged VLANs: 30
  Untagged VLANs: 30
  Untagged VLANs: 30
  Untagged VLANs: 1
  Tagged VLANs: 30
  Tagged VLANs: 35
  Tagged VLANs: 200
########################
``` 
but I'd like to have this rendering, a list of vlan with unique ids, name and description

**Rendered Config Needed**
```
########################
vlan 1
name vlname-id1
description vlandec-id1

vlan 30
name vlname-id30
description vlandec-id30

vlan 35
name vlname-id35
description vlandec-id35

vlan 200
name vlname-id200
description vlandec-id200
########################
```
I can't use this variable "_device.site.vlans.all()_" because vlans can be present on several sites but not all.

Can you help me, please?

**Comment by pl0xym0r** [comment]
maybe not the fastest way but : 

- create a list that appends every vlan you need, then  loop over `| unique` to only have vlan vid once.
- **OR** `do` and/or `set` jinja instructions to filter what you need
- **OR**  code you own jinja filter in python and apply it

**Comment by csmith35-ctd** [comment]
i try this way
```jinja
########################
{%- set vlList = [] -%}
{%- for interface in device.interfaces.all() -%}
{%- if interface.enabled == true and interface.untagged_vlan.vid != "None" and interface.type != "virtual" -%}{%- set _ = vlList.append(interface.untagged_vlan.vid) -%}
  {%- for vlan in interface.tagged_vlans.all() -%}
  {%- if vlan.id != "" -%}
  {%- set _ = vlList.append(vlan.vid) -%}
  {%- endif -%}{%- endfor -%}
{%- endif -%}{%- endfor -%}
{%- for vlan in vlList|sort|unique %}
vlan {{vlan}}
name
description 
{% endfor -%}
########################
``` 
**Rendered Config**
``` 
########################
vlan 1
name
description

vlan 30
name
description

vlan 35
name
description

vlan 200
name
description
########################
``` 
I don't see how i can get the name and description for each vlan.

**Reply by pl0xym0r** [reply]
You only append vid in your list, you could append the object or the id to get the object after

**Comment by csmith35-ctd** [comment]
Thx you very much, I found and succeeded.
i used the id to get the objet after in ipam.VLAN.objects.all()

**Reply by pl0xym0r** [reply]
well, you could filter directly like ipam.VLAN.objects.filter(vid_in=vlList) and loop over it directly instead of parse .all()

{%- for vlan in ipam.VLAN.objects.filter(vid_in=vlList).distinct() %}
vlan {{vlan.vid}}
name  {{vlan.name}}
{% endfor -%}

**Comment by csmith35-ctd** [comment, marked as answer]
Thx, very nice filter
i just have to add an **_** on vid__in, and that's works fine

`{%- for vlan in ipam.VLAN.objects.filter(vid__in=vlList).distinct() %}`
